llvm-general/test/LLVM/General/Test/Linking.hs

```diff
@@ -59,9 +59,9 @@ tests = testGroup "Linking" [
         ]      
 
     Module { moduleDefinitions = defs } <- withContext $ \context -> 
-      withModuleFromAST' context ast0 $ \m0 ->
-        withModuleFromAST' context ast1 $ \m1 -> do
-          runExceptT $ linkModules False m0 m1
-          moduleAST m0
+      withModuleFromAST' context ast0 $ \dest -> do
+        src <- failInIO $ astToFFIModule context ast1
+        failInIO $ linkModules dest src
```

Comment:
We don’t need this change in 3.8 because we are using the old linking function there (which got removed in 3.9). I am not going to backport this change but feel free to do so yourself.